ably/rest_channel.go

```diff
@@ -1,24 +1,21 @@
-package rest
+package ably
 
-import (
-	"github.com/ably/ably-go/config"
-	"github.com/ably/ably-go/proto"
-)
+import "github.com/ably/ably-go/ably/proto"
 
-type Channel struct {
+type RestChannel struct {
 	Name     string
-	Presence *Presence
+	Presence *RestPresence
 
-	client *Client
+	client *RestClient
 }
 
-func newChannel(name string, client *Client) *Channel {
-	c := &Channel{
+func newRestChannel(name string, client *RestClient) *RestChannel {
+	c := &RestChannel{
 		Name:   name,
 		client: client,
 	}
 
-	c.Presence = &Presence{
+	c.Presence = &RestPresence{
```

Comment:
Can you add to the other issue that `Presence` needs to be a getter (`presence()`). This is to restrict the access to the `Presence` object.